### PR TITLE
Refactoring `alert()` macro

### PR DESCRIPTION
**Pull request by cavasinf**
## Description
Related to #89 

#### Usage

```twig
{% from '@Tabler/components/alert.html.twig' import alert %}

<div class="row">
    <div class="col-6">
        {{ alert('danger', 'Description' ,'Title', 'fas fa-exclamation-triangle', true) }}
        {{ alert('danger', 'Description' ,'Title', 'fas fa-exclamation-triangle') }}
        {{ alert('danger', 'Description' ,'Title', 'fas fa-exclamation-triangle') }} {# Error here #}
        {{ alert('warning', 'Description' ,'Title') }}
        {{ alert('warning', null ,'Title') }}
        {{ alert('warning', null ,'Title', 'fas fa-exclamation-triangle') }}
        {{ alert('success', 'Description') }}
        {{ alert('success', 'Description', null, 'fas fa-exclamation-triangle') }}
        {{ alert('primary') }}
    </div>

    <div class="col-6">
        {{ alert({title : 'Title V2', description: 'Description V2', icon : 'warning', dismissible : true}) }}
        {{ alert({title : 'Title V2', description: 'Description V2', icon : 'warning'}) }}
        {{ alert({title : 'Title V2', description: 'Description V2', icon : 'warning', important: true}) }} {# Looks good now #}
        {{ alert({title : 'Title V2', description: 'Description V2', type : 'warning'}) }}
        {{ alert({title : 'Title V2', type : 'warning'}) }}
        {{ alert({title : 'Title V2', icon : 'warning', type : 'warning'}) }}
        {{ alert({description : 'Description V2', type : 'success'}) }}
        {{ alert({description : 'Description V2', type : 'success', icon : 'warning'}) }}
        {{ alert({type : 'primary'}) }}
    </div>
</div>
```

#### Preview

image

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [ ] I updated the documentation (see [here](https://github.com/kevinpapst/TablerBundle/tree/master/docs))
- [x] I agree that this code will be published under the [MIT license](https://github.com/kevinpapst/TablerBundle/blob/master/LICENSE)

### CODE REVIEW

File: templates/components/alert.html.twig [hunk 1]
```diff
@@ -1,27 +1,51 @@
...
+    {% if type is iterable %}
+        {% set options = type %}
+
+        {% set _type = options.type ?? 'danger' %}
+        {% set _title = options.title ?? null %}
+        {% set _description = options.description ?? null %}
+        {% set _icon = options.icon ?? null %}
+        {% set _dismissible = (options.dismissible ?? false) is same as (true) %}
+        {% set _important = (options.important ?? false) is same as (true) %}
+
```
Comment (kevinpapst) [inline templates/components/alert.html.twig:12]: {% set _raw = (options.raw ?? false) is same as (true) %}

File: templates/components/alert.html.twig [hunk 2]
```diff
@@ -1,27 +1,51 @@
...
+                <div class="alert-icon">
+                    {{ tabler_icon(_icon, true) }}
+                </div>
+            {% endif %}
+
+            {% if _title is not empty or _description is not empty %}
                 <div>
-                    <i class="{% if background %}icon{% else %}alert-icon{% endif%} {{ icon|tabler_icon(false, icon) }}"></i>
+                    {% if _title is not empty %}
+                        <h3 class="{{ _description is not empty ? 'mb-1' : 'mb-0' }}">{{ _title|raw }}</h3>
```
Comment (kevinpapst) [inline templates/components/alert.html.twig:40]: {% if _raw %}{{ _title|raw }}{% else %}{{ _title }}{% endif %}

**Review by kevinpapst — CHANGES_REQUESTED** [review]
Can you make the raw filter optional by a setting like in my example.
This filter can easily lead to CSRF issues if you pass user generated values to it. 
Better activate it per case, instead of "blindly" using unescaped data.

**Comment by cavasinf** [comment]
Everything's OK for me 👌

**Comment by kevinpapst** [comment]
I will add your examples to the demo application. Thanks you @cavasinf 👍

**Comment by cavasinf** [comment]
Yeah I haven't seen a doc for that kind of feature.
We can maybe add the doc of that component is this repo like others 🤔

**Comment by kevinpapst** [comment]
This is what I see after the merge:
Bildschirmfoto 2022-06-01 um 09 38 27
Can you confirm that is still looks as in your screenshot?

**Comment by kevinpapst** [comment]
> Yeah I haven't seen a doc for that kind of feature.

Because you still don't use the demo app, do you 😁

**Comment by kevinpapst** [comment]
Okay, there is one BC break which is not matching the Tabler docs: https://preview.tabler.io/docs/alerts.html

Before:
Bildschirmfoto 2022-06-01 um 09 45 33
 
After:
Bildschirmfoto 2022-06-01 um 09 51 29

Could you have a look while I work on the docs?

**Comment by cavasinf** [comment]
@kevinpapst Looks like it's coming from the `icon` class again.

Tabler user SVG, we use FontAwesome.
Can you try with `false` param at `tabler_icon` twig function.
https://github.com/kevinpapst/TablerBundle/blob/da494e86cab2d04f5ea500ab04850cb918d8a675/templates/components/alert.html.twig#L34

I remember it was "odd" without this class, but if it's okay for you, we can switch that 👍

**Comment by cavasinf** [comment]
I'll check for title color + description => new PR ?

**Comment by kevinpapst** [comment]
I don't mind the icon size/positioning, bigger looks better and I didn't even recognize it before you mentioned it ;-)

I was only talking about the colors, which worked before. New PR sounds good :)

**Comment by kevinpapst** [comment]
Going through my deprecations log, I saw that we missed one: 
https://github.com/kevinpapst/TablerBundle/blob/main/templates/components/callout.html.twig#L3

`callout` is calling `alert` with single params 🙄

**Comment by cavasinf** [comment]
What do you want to do ? 
Add the deprecated message and change multi params into object one param

**Comment by kevinpapst** [comment]
Because important is the 6 param and it got lost somewhere in the refactoring process.
So have a consistent API we should change the signature as well:
```
{% from '@theme/components/alert.html.twig' import alert %}
{% if type is iterable %}
    {{ alert(type) }}
{% else %}
    {% deprecated "Passing multiple parameters to 'callout()' Tabler macro is deprecated, use options object syntax." %}
    {{ alert({type: type|default('danger'), description: description, title: title|trans, icon: icon, dismissible: false, important: true}) }}
{% else %}
```

What do you think ?

**Comment by cavasinf** [comment]
Missing the trans of `description` (like OLD alert macro) but LGTM 👍

**Comment by cavasinf** [comment]
- Missing the trans of `description` (like OLD alert macro)
- `icon` is required in callout macro ?
- `description` is required ?
- Maybe add `{% set options = type %}` inside if condition (if devs looks at code).

**Pull request merged**